nntrainer/src/neuralnet.cpp

```diff
@@ -455,7 +455,7 @@ int NeuralNetwork::train_run() {
       if (data_buffer->getDataFromBuffer(nntrainer::BufferType::BUF_TRAIN,
                                          in->getData(), label->getData())) {
         try {
-          backwarding({in}, {label}, iter++);
+          backwarding({in}, {label}, epoch_idx);
```

Comment:
This is wrong. This loop runs for multiple iterations and backwarding needs that iteration count to apply the gradient.
This update passes epoch count which remains the same for all the iterations in the epoch.